lib/Compiler.cpp

```diff
@@ -1394,6 +1398,12 @@ ClspvError clspvCompileFromSourcesString(
 
   // Alloc and copy backing mem for spv output
   size_t spv_bytes = binary.size() * sizeof(uint32_t);
+  if (!spv_bytes) {
+    /** Early return: when allocation is not needed. */
+    *spv_bytes = static_cast<void *>(NULL);
+    return CLSPV_SUCCESS;
+  }
```

Comment:
```suggestion
  if (spv_bytes == 0) {
    /** Early return: when allocation is not needed. */
    *output_binary = static_cast<char *>(NULL);
    return CLSPV_SUCCESS;
  }
```
`spv_bytes` is a `size_t` not a boolean, let's keep thing as readable as possible by avoiding boolean operators on `size_t` (even so I understand that it produced the same outcome here).

`*spv_bytes` feels like a bug as `spv_bytes` is a `size_t` not a pointer.

Reply:
`spv_bytes` is a bug. changed to `output_binary`.